src/graphics/vulkan.zig

```diff
@@ -156,3 +179,111 @@ pub fn createInstance() void {
 	};
 	checkResult(c.vkCreateInstance(&createInfo, null, &instance));
 }
+
+// MARK: Physical Device
+
+const deviceExtensions = [_][*:0]const u8{
+	c.VK_KHR_SWAPCHAIN_EXTENSION_NAME,
+};
+
+const QueueFamilyIndidices = struct {
+	graphicsFamily: ?u32 = null,
+	presentFamily: ?u32 = null,
+
+	fn isComplete(self: QueueFamilyIndidices) bool {
+		return self.graphicsFamily != null and self.presentFamily != null;
+	}
+};
+
+fn findQueueFamilies(dev: c.VkPhysicalDevice) QueueFamilyIndidices {
+	var result: QueueFamilyIndidices = .{};
+	const queueFamilies = getPhysicalDeviceQueueFamilyProperties(main.stackAllocator, dev);
+	defer main.stackAllocator.free(queueFamilies);
+	for(queueFamilies, 0..) |family, i| {
+		if(family.queueFlags & c.VK_QUEUE_GRAPHICS_BIT != 0) {
+			result.graphicsFamily = @intCast(i);
+		}
+		var presentSupport: u32 = 0;
+		checkResult(c.vkGetPhysicalDeviceSurfaceSupportKHR(dev, @intCast(i), surface, &presentSupport));
+		if(presentSupport != 0) {
+			result.presentFamily = @intCast(i);
+		}
+	}
+	return result;
+}
+
+fn checkDeviceExtensionSupport(dev: c.VkPhysicalDevice) bool {
+	const availableExtension = enumerateDeviceExtensionProperties(main.stackAllocator, dev, null);
+	defer main.stackAllocator.free(availableExtension);
+	for(deviceExtensions) |requiredName| continueOuter: {
+		for(availableExtension) |available| {
+			if(std.mem.eql(u8, std.mem.span(requiredName), std.mem.span(@as([*:0]const u8, @ptrCast(&available.extensionName))))) {
+				break :continueOuter;
+			}
+		}
+		std.log.warn("Rejecting device because extension {s} was not found", .{requiredName});
+		return false;
+	}
+	return true;
+}
+
+fn deviceScore(dev: c.VkPhysicalDevice) f32 {
```

Comment:
```suggestion
fn getDeviceScore(dev: c.VkPhysicalDevice) f32 {
```

Start function name with a verb as function is an action to perform.